### PR TITLE
Move `FileClient` and `BlockFileCodec` out of test-utils

### LINKED ISSUE #5994: Move FileClient and BlockFileCodec out of test-utils
### Describe the feature

These are used in `reth import`, which is used actively in hive, so these should be brought into either a new crate or somewhere else that is not gated behind the `test-utils` feature

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #5994.

### CODE REVIEW

File: crates/net/downloaders/src/file_client.rs [hunk 1]
```diff
@@ -127,12 +127,14 @@ impl FileClient {
     }
 
     /// Use the provided bodies as the file client's block body buffer.
+    #[allow(dead_code)]
     pub(crate) fn with_bodies(mut self, bodies: HashMap<BlockHash, BlockBody>) -> Self {
```
Comment (rkrasiuk) [inline crates/net/downloaders/src/file_client.rs:131]: if we were to use these methods in tests only, they should have been`#[cfg(test)]`, but let's make these `pub` instead

Comment (tcoratger) [reply crates/net/downloaders/src/file_client.rs]: Fine, just did it.

File: crates/net/downloaders/src/lib.rs [hunk 2]
```diff
@@ -20,5 +20,16 @@ pub mod headers;
 /// Common downloader metrics.
 pub mod metrics;
 
+/// Module managing file-based data retrieval and buffering.
+///
+/// Contains `FileClient` to read block data from files,
+/// efficiently buffering headers and bodies for retrieval.
+pub mod file_client;
+
+/// Module with a codec (`BlockFileCodec`) for reading and encoding block bodies in files.
```
Comment (Rjected) [inline crates/net/downloaders/src/lib.rs:29]:
```suggestion
/// Contains [FileClient] to read block data from files,
/// efficiently buffering headers and bodies for retrieval.
pub mod file_client;

/// Module with a codec [BlockFileCodec] for reading and encoding block bodies in files.
```

**Review by Rjected — CHANGES_REQUESTED** [review]
Just one nit about docs, otherwise this looks good to me!

It should also be possible to remove the `test-utils` feature from `bin/reth/Cargo.toml`

**Comment by Rjected** [comment]
whoops, should have been `[FileClient](file_client::FileClient)` and `[BlockFileCodec](file_codec::BlockFileCodec)`

**Comment by tcoratger** [comment]
> whoops, should have been `[FileClient](file_client::FileClient)` and `[BlockFileCodec](file_codec::BlockFileCodec)`

@Rjected Yep just fixed it plus Cargo dependencies.

**Review by Rjected — APPROVED** [review]
LGTM, thanks!

**Pull request merged**